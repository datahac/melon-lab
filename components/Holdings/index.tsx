import React, { useState } from 'react';
import Holdings from '~/components/Holdings';
import Router from 'next/router';
import * as R from 'ramda';
import tokens from '~/utils/tokens';
import ReleaseToken from '../ReleaseToken';

const mapHoldings = R.curryN(2, (nav, asset) => {
  const token = R.find(R.propEq('key', R.path(['balance', 'token', 'symbol'], asset)), tokens);

  return {
    price: asset.price,
    balance: asset.balance,
    locked: asset.locked,
    name: token && token.value,
    symbol: asset.balance.token.symbol,
    address: asset.balance.token.address,
  };
});

const sortHoldings = R.sortWith([R.descend(R.path(['balance', 'quantity'])), R.ascend(R.prop('symbol'))]);

const HoldingsContainer = ({ nav, loading, address, isManager, holdings, baseAsset, quoteAsset }) => {
  const [releaseToken, setReleaseToken] = useState(null);

  const onClick = (asset, address) => {
    Router.push({
      pathname: '/manage',
      query: {
        address,
        base: asset.symbol,
        quote: 'WETH',
      },
    });
  };

  return (
    <>
      <ReleaseToken fundAddress={address} releaseToken={releaseToken} setReleaseToken={setReleaseToken} />
      <Holdings
        quoteAsset={quoteAsset}
        baseAsset={baseAsset}
        holdings={sortHoldings(holdings.map(mapHoldings(nav)))}
        loading={loading}
        isManager={isManager}
        onClick={asset => onClick(asset, address)}
        setReleaseToken={setReleaseToken}
      />
    </>
  );
};

export default HoldingsContainer;
