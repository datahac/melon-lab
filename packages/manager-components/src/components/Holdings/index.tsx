import React, { Fragment, StatelessComponent } from 'react';
import Button from '~/blocks/Button';
import Loading from '~/blocks/Loading';
import {
  CellBody,
  CellHead,
  Row,
  Table,
  TableBody,
  TableHead,
} from '~/blocks/Table';

import styles from './styles.css';

export interface Holding {
  balance: string;
  name: string;
  percentage: string;
  price: string;
}

export interface HoldingsProps {
  dataValid?: boolean;
  holdings?: Holding[];
  isReadyToTrade?: boolean;
  quoteAsset?: string;
  selectAsset?: (assetName, isReadyToTrade, quoteAsset) => void;
}

export const Holdings: StatelessComponent<HoldingsProps> = ({
  dataValid,
  holdings,
  isReadyToTrade,
  quoteAsset,
  selectAsset,
}) => {
  const onClick = (e: any, assetName: string): void =>
    selectAsset && selectAsset(assetName, isReadyToTrade, quoteAsset);

  return (
    <div className="holdings" id="holdings">
      <style jsx>{styles}</style>
      <h3>Fund Holdings</h3>
      <div className="holdings__table-wrap">
        <Table>
          <TableHead>
            <Row isHead={true}>
              <CellHead>Asset</CellHead>
              <CellHead>Quantity</CellHead>
              <CellHead textAlign="right">% of portfolio</CellHead>
              <CellHead textAlign="right">Price (in {quoteAsset})</CellHead>
              <CellHead textAlign="right" />
            </Row>
          </TableHead>
          <TableBody>
            {holdings &&
              holdings.map(asset => (
                <Row key={asset.name} size="small">
                  <CellBody>{asset.name}</CellBody>
                  <CellBody>{asset.balance}</CellBody>
                  <CellBody textAlign="right">
                    {dataValid ? (
                      <Fragment>{asset.percentage}%</Fragment>
                    ) : (
                      <Loading />
                    )}
                  </CellBody>
                  <CellBody textAlign="right">
                    {dataValid ? (
                      <Fragment>{asset.price}</Fragment>
                    ) : (
                      <Loading />
                    )}
                  </CellBody>
                  <CellBody textAlign="right" cellClass="holdings__action-cell">
                    {asset.name === quoteAsset ? (
                      <span className="holdings__quote-asset" />
                    ) : asset.name !== quoteAsset &&
                    isReadyToTrade &&
                    dataValid ? (
                      <Button
                        size="small"
                        buttonValue={asset.name}
                        onClick={onClick}
                        style="secondary"
                      >
                        Buy/Sell
                      </Button>
                    ) : (
                      <Button
                        size="small"
                        buttonValue={asset.name}
                        onClick={onClick}
                        style="secondary"
                      >
                        See Orderbook
                      </Button>
                    )}
                  </CellBody>
                </Row>
              ))}
          </TableBody>
        </Table>
      </div>
    </div>
  );
};

export default Holdings;
