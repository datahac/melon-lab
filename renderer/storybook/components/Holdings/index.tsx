import React, { StatelessComponent } from 'react';
import Spinner from '~/blocks/Spinner';
import Holding from '~/blocks/Holding';
import * as Tm from '@melonproject/token-math';

import styles from './styles.css';

export interface Holding {
  balance: Tm.QuantityInterface;
  symbol: string;
  fraction: string;
  price: Tm.PriceInterface;
}

export interface HoldingsProps {
  holdings?: Holding[];
  quoteAsset?: string;
  baseAsset?: string;
  loading?: boolean;
  onClick: (asset) => void;
}

export const Holdings: StatelessComponent<HoldingsProps> = ({
  holdings,
  quoteAsset,
  baseAsset,
  loading,
  onClick,
}) => (
  <div className="holdings" id="holdings">
    <style jsx>{styles}</style>
    {loading ? (
      <div className="holdings__loading">
        <Spinner icon size="small" />
      </div>
    ) : (
      <div className="holdings__table-wrap">
        {holdings &&
          holdings.map(asset => (
            <Holding
              key={asset.symbol}
              fraction={asset.fraction}
              symbol={asset.symbol}
              name={asset.name}
              price={asset.price}
              balance={asset.balance}
              active={baseAsset === asset.symbol}
              onClickHolding={
                !(asset.symbol === quoteAsset) ? () => onClick(asset) : null
              }
            />
          ))}
      </div>
    )}
  </div>
);

export default Holdings;
