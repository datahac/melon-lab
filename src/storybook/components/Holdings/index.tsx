import React, { StatelessComponent } from 'react';
import Spinner from '~/blocks/Spinner';
import Holding from '~/blocks/Holding';
import * as Tm from '@melonproject/token-math';

import styles from './styles.css';
import Table, { TableHead, Row, CellHead, TableBody } from '~/blocks/Table';

export interface Holding {
  balance: Tm.QuantityInterface;
  price: Tm.PriceInterface;
  symbol: string;
  fraction: string;
  name: string;
  address: string;
}

export interface HoldingsProps {
  holdings?: Holding[];
  quoteAsset?: string;
  baseAsset?: string;
  loading?: boolean;
  isManager: boolean;
  setReleaseToken: (asset) => void;
  onClick: (asset) => void;
}

export const Holdings: StatelessComponent<HoldingsProps> = ({
  holdings,
  quoteAsset,
  baseAsset,
  loading,
  isManager,
  setReleaseToken,
  onClick,
}) => (
  <div className="holdings" id="holdings">
    <style jsx>{styles}</style>
    {loading ? (
      <div className="holdings__loading">
        <Spinner icon size="small" />
      </div>
    ) : (
      <div className="holdings__table-wrap">
        <Table>
          <TableHead>
            <Row isHead={true}>
              <CellHead noPadding={false}>Asset</CellHead>
              <CellHead noPadding={false}>Price</CellHead>
              <CellHead noPadding={false}>Balance</CellHead>
            </Row>
          </TableHead>
          <TableBody>
            {holdings &&
              holdings.map(asset => (
                <Holding
                  key={asset.symbol}
                  fraction={asset.fraction}
                  symbol={asset.symbol}
                  name={asset.name}
                  isManager={isManager}
                  price={asset.price}
                  balance={asset.balance}
                  locked={asset.locked}
                  active={baseAsset === asset.symbol}
                  onClickHolding={!(asset.symbol === quoteAsset) ? () => onClick(asset) : null}
                  onClickRelease={() => setReleaseToken(asset.address)}
                />
              ))}
          </TableBody>
        </Table>
      </div>
    )}
  </div>
);

export default Holdings;
