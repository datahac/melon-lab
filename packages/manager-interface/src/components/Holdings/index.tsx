import Holdings from '~/components/Holdings';
import { compose, withPropsOnChange, withHandlers } from 'recompose';
import * as R from 'ramda';
import Router from 'next/router';
import { toBigNumber } from '~/utils/functionalBigNumber';

// TODO: Add isReadyToTrade

const withHoldingHandlers = withHandlers({
  onClick: props => asset => {
    Router.push({
      pathname: '/manage',
      query: { address: props.address, base: asset.symbol, quote: 'WETH-T' },
    });
  },
});

const mapHoldings = R.curryN(2, (nav, asset) => ({
  ...asset,
  price: asset.price,
  fraction: toBigNumber(asset.balance)
    .times(asset.price)
    .div(nav.toString() || 1)
    .times(100),
  balance: asset.balance,
}));

const withMappedProps = withPropsOnChange(
  ['holdings', 'nav', 'loading'],
  props => ({
    holdings:
      (!props.loading &&
        props.holdings &&
        props.holdings.map(mapHoldings(props.nav))) ||
      [],
  }),
);

export default compose(
  withMappedProps,
  withHoldingHandlers,
)(Holdings);
