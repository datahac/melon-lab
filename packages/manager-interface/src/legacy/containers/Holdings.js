import { connect } from 'react-redux';
import { lifecycle } from 'recompose';
import { actions } from '../actions/holdings';
import { actions as appActions } from '../actions/app';
import { actions as orderbookActions } from '../actions/orderbook';
import { actions as recentTradesActions } from '../actions/recentTrades';
import Holdings from '@melonproject/manager-components/components/Holdings';
import displayNumber from '../utils/displayNumber';

const mapStateToProps = state => ({
  holdings: state.holdings.holdings.map(asset => ({
    name: asset.name,
    balance: displayNumber(asset.balance),
    price: displayNumber(asset.price),
    percentage:
      state.fund.nav === '...'
        ? '...'
        : displayNumber(
            asset.balance
              .times(asset.price)
              .div(state.fund.nav || 1)
              .times(100),
          ),
  })),
  dataValid: state.ethereum.isDataValid,
  isReadyToTrade: state.app.isReadyToTrade,
  quoteAsset: state.app.assetPair.quote,
});

const mapDispatchToProps = dispatch => ({
  getHoldings: fundAddress => {
    dispatch(actions.getHoldings(fundAddress));
  },
  selectAsset: (asset, isReadyToTrade, quoteAsset) => {
    if (asset !== quoteAsset) {
      dispatch(appActions.updateAssetPair({ base: asset }));
      dispatch(orderbookActions.getOrderbook());
      dispatch(recentTradesActions.getRecentTrades());
      if (isReadyToTrade) {
        dispatch(appActions.scrollTo('trade'));
      } else {
        dispatch(appActions.scrollTo('orderbook'));
      }
    }
  },
});

const HoldingsWithLifecycle = lifecycle({
  componentDidMount() {
    this.props.getHoldings(this.props.fundAddress);
  },
})(Holdings);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(HoldingsWithLifecycle);
