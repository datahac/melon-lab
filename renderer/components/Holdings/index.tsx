import Holdings from '~/components/Holdings';
import Router from 'next/router';
import * as R from 'ramda';
import tokens from '~/shared/utils/tokens';

const mapHoldings = R.curryN(2, (nav, asset) => {
  const token = R.find(R.propEq('key', asset.balance.token.symbol), tokens);

  return {
    price: asset.price,
    // TODO: Re-implement fraction.
    fraction: 0,
    balance: asset.balance,
    name: token && token.value,
    symbol: asset.balance.token.symbol,
  };
});

const sortHoldings = R.sortWith([
  R.descend(R.prop('fraction')),
  R.ascend(R.prop('symbol')),
]);

export default class HoldingsContainer extends React.PureComponent {
  onClick(asset, address) {
    Router.push({
      pathname: '/manage',
      query: {
        address,
        base: asset.symbol,
        quote: 'WETH',
      },
    });
  }

  render() {
    const {
      nav,
      loading,
      address,
      holdings,
      baseAsset,
      quoteAsset,
    } = this.props;

    return (
      <Holdings
        quoteAsset={quoteAsset}
        baseAsset={baseAsset}
        holdings={sortHoldings(holdings.map(mapHoldings(nav)))}
        loading={loading}
        onClick={asset => this.onClick(asset, address)}
      />
    );
  }
}
