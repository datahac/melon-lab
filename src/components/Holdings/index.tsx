import Holdings from '~/components/Holdings';
import Router from 'next/router';
import * as R from 'ramda';
import { toBigNumber } from '~/utils/functionalBigNumber';
import tokens from '~/utils/tokens';

const mapHoldings = R.curryN(2, (nav, asset) => {
  return {
    ...asset,
    price: asset.price,
    fraction:
      asset &&
      nav &&
      toBigNumber(asset.balance)
        .times(asset.price)
        .div(nav.toString() || 1)
        .times(100),
    balance: asset.balance,
    tokenName: R.pathOr(
      'N/A',
      ['value'],
      R.find(R.propEq('key', asset.symbol.split('-')[0]), tokens),
    ),
    tokenSymbol: asset.symbol,
  };
});

const sortHoldings = R.sortWith([
  R.comparator((a, b) =>
    R.gt(parseInt(R.prop('fraction', a)), parseInt(R.prop('fraction', b))),
  ),
  R.ascend(R.prop('symbol')),
]);

export default class HoldingsContainer extends React.PureComponent {
  onClick(asset, address) {
    Router.push({
      pathname: '/manage',
      query: {
        address,
        base: asset.symbol,
        quote: 'WETH-T',
      },
    });
  }

  render() {
    const { address, baseAsset, quoteAsset } = this.props;
    const holdings = sortHoldings(
      this.props.holdings.map(mapHoldings(this.props.nav)),
    );

    return (
      <Holdings
        quoteAsset={quoteAsset}
        baseAsset={baseAsset}
        holdings={holdings}
        loading={this.props.loading}
        onClick={asset => this.onClick(asset, address)}
      />
    );
  }
}
