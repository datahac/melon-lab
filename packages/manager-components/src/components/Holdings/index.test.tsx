import React from 'react';
import Holdings from './index';

const data = {
  isReadyToTrade: true,
  dataValid: true,
  quoteAsset: 'WETH-T',
  holdings: [
    { name: 'ANT-T', balance: '0.0000', price: '0.0035', percentage: '0.0000' },
    { name: 'BAT-T', balance: '0.0000', price: '0.0007', percentage: '0.0000' },
    { name: 'DGD-T', balance: '0.0000', price: '0.0000', percentage: '0.0000' },
    { name: 'DGX-T', balance: '0.0000', price: '0.0000', percentage: '0.0000' },
    { name: 'GNO-T', balance: '0.0000', price: '0.1113', percentage: '0.0000' },
    { name: 'REP-T', balance: '0.0000', price: '0.0648', percentage: '0.0000' },
    { name: 'ZRX-T', balance: '0.0000', price: '0.0024', percentage: '0.0000' },
    { name: 'REQ-T', balance: '0.0000', price: '0.0001', percentage: '0.0000' },
    {
      name: 'WETH-T',
      balance: '0.5166',
      price: '1.0000',
      percentage: '50.5764',
    },
    {
      name: 'MLN-T',
      balance: '10.7338',
      price: '0.0470',
      percentage: '49.4236',
    },
    { name: 'MKR-T', balance: '0.0000', price: '1.3430', percentage: '0.0000' },
    { name: 'DAI-T', balance: '0.0000', price: '0.0021', percentage: '0.0000' },
    { name: 'KNC-T', balance: '0.0000', price: '0.0276', percentage: '0.0000' },
    { name: 'JNT-T', balance: '0.0000', price: '0.0003', percentage: '0.0000' },
    { name: 'OMG-T', balance: '0.0000', price: '0.0150', percentage: '0.0000' },
    { name: 'NMR-T', balance: '0.0000', price: '0.0184', percentage: '0.0000' },
  ],
};

describe('Holdings', () => {
  const defaultElement = <Holdings {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
