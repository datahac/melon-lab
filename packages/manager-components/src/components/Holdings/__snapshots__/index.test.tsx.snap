// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Holdings should render correctly 1`] = `
<div
  className="jsx-__scopedHash holdings"
  id="holdings"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Fund Holdings
  </h3>
  <div
    className="jsx-__scopedHash holdings__table-wrap"
  >
    <Table>
      <TableHead>
        <Row
          isHead={true}
        >
          <CellHead>
            Asset
          </CellHead>
          <CellHead>
            Quantity
          </CellHead>
          <CellHead
            textAlign="right"
          >
            % of portfolio
          </CellHead>
          <CellHead
            textAlign="right"
          >
            Price (MLN)
          </CellHead>
          <CellHead
            textAlign="right"
          />
        </Row>
      </TableHead>
      <TableBody>
        <Row
          key="ANT-T"
          size="small"
        >
          <CellBody>
            ANT-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0035
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="ANT-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="BAT-T"
          size="small"
        >
          <CellBody>
            BAT-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0007
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="BAT-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="DGD-T"
          size="small"
        >
          <CellBody>
            DGD-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="DGD-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="DGX-T"
          size="small"
        >
          <CellBody>
            DGX-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="DGX-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="GNO-T"
          size="small"
        >
          <CellBody>
            GNO-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.1113
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="GNO-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="REP-T"
          size="small"
        >
          <CellBody>
            REP-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0648
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="REP-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="ZRX-T"
          size="small"
        >
          <CellBody>
            ZRX-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0024
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="ZRX-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="REQ-T"
          size="small"
        >
          <CellBody>
            REQ-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0001
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="REQ-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="WETH-T"
          size="small"
        >
          <CellBody>
            WETH-T
          </CellBody>
          <CellBody>
            0.5166
          </CellBody>
          <CellBody
            textAlign="right"
          >
            50.5764
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            1.0000
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <span
              className="jsx-__scopedHash holdings__quote-asset"
            />
          </CellBody>
        </Row>
        <Row
          key="MLN-T"
          size="small"
        >
          <CellBody>
            MLN-T
          </CellBody>
          <CellBody>
            10.7338
          </CellBody>
          <CellBody
            textAlign="right"
          >
            49.4236
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0470
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="MLN-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="MKR-T"
          size="small"
        >
          <CellBody>
            MKR-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            1.3430
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="MKR-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="DAI-T"
          size="small"
        >
          <CellBody>
            DAI-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0021
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="DAI-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="KNC-T"
          size="small"
        >
          <CellBody>
            KNC-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0276
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="KNC-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="JNT-T"
          size="small"
        >
          <CellBody>
            JNT-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0003
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="JNT-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="OMG-T"
          size="small"
        >
          <CellBody>
            OMG-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0150
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="OMG-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="NMR-T"
          size="small"
        >
          <CellBody>
            NMR-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0000
            %
          </CellBody>
          <CellBody
            textAlign="right"
          >
            0.0184
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
            textAlign="right"
          >
            <Button
              buttonValue="NMR-T"
              onClick={[Function]}
              size="small"
              style="secondary"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
      </TableBody>
    </Table>
  </div>
</div>
`;
